GSA-TTS/FAC
# Enforce Expiration Date for UEI Validation Waivers

# Enforce Expiration Date for UEI Validation Waivers

Issue: https://github.com/GSA-TTS/FAC/issues/3978, checklist item number 6

## Changes:
1. Set the default expiration date for a UEI validation waiver at 30 days after the current time
2. Adjust UEIValidationWaiver model so that the admin panel can enter and display date fields
    * Includes a migration to adjust default times
3. Enforce expiration dates in UEI validation


## How to test:
1. Switch to this branch and run normally
2. Add two UEI validation waivers via the admin panel - one that is valid, and one that is expired
    * I used `SUPERC00LUE1` and `0LDANDSADUE1`
4. Ensure that the UEIs are accepted or denied based on their expiration status


## Recording:

https://github.com/user-attachments/assets/365d053b-2f2a-4f46-b43c-1ad5b0437e06


## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
